esp-rtos/src/lib.rs

```diff
@@ -242,6 +242,7 @@ pub fn start_second_core<const STACK_SIZE: usize>(
     #[cfg(xtensa)] int0: SoftwareInterrupt<'static, 0>,
     int1: SoftwareInterrupt<'static, 1>,
     stack: &'static mut Stack<STACK_SIZE>,
+    stack_guard_offset: Option<usize>,
```

Comment:
I would prefer not having this argument. The stack guard's position effectively determines the end of the stack, anything more is unuseable. If we touch the stack guard in any way, the program crashes with this PR. I think it should be fine to just place the stack guard at the very end of the stack.

Reply:
In my experience placing the guard at the very end of the stack is very likely to trigger after memory below is already touched - I never had good results with an offset of 0

Might or might not be as much of a problem as with the core0 main stack but while it might trigger we are often not able to even log anything anymore

Having the safe-area at the bottom of the stack certainly means giving up on some of the available memory and there is no one-size fits all unfortunately - having this as an argument at least leaves passing `Some(0)` to the brave users and passing something else to others

Reply:
Sounds interesting, let's try this.